api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -606,4 +606,16 @@
      */
     public <T extends Extension> T getExtension(Class<T> extensionClass);
 
+    /**
+     *
+     * Create an {@link InterceptionProxyFactory} for the given {@link CreationalContext} and type.
+     *
+     * @param ctx {@link CreationalContext} for the {@link InterceptionProxyFactory} to create
+     * @param clazz class of the instance to apply proxies on
+     * @param <T> type of the instance to apply proxies on
+     * @return a new {@link InterceptionProxyFactory} to create proxies for an instance of T
+     * @since 2.0
+     */
+    <T> InterceptionProxyFactory<T> createInterceptionFactory(CreationalContext<T> ctx, Class<T> clazz);
```

Comment:
If anyone can refresh my mind, I know we talked about built-in beans, but what was the reason behind a BM method? I would guess Extensions, is that right?